### PR TITLE
fix: use base node12 tsconfig

### PR DESCRIPTION
**Pull request by SimenB**
## About the changes

The current transpile target transpiles a bunch of code that's not needed, such as `async` functions and `class`es - both of which are heavily used in this code base.

You target a minimum of node 12, so that's the config I've used: https://github.com/Unleash/unleash-client-node/blob/18ee8ef5a397b8d1615dccd1bbc93edea85bfa0c/.github/workflows/master.yaml#L13

### Important files

`tsconfig.json`

## Discussion points

This is the full diff of the transpiled code: https://gist.github.com/SimenB/6608476192348cc2676dd75e6d76cdf8

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,17 +1,14 @@
 {
+  "extends": "@tsconfig/node12/tsconfig.json",
   "compilerOptions": {
     "rootDir": "./src",
     "outDir": "./lib",
-    "lib": ["es5", "es6"],
-    "target": "es5",
-    "module": "commonjs",
```
Comment (SimenB) [inline tsconfig.json:7]: comes from extended config

File: tsconfig.json [hunk 2]
```diff
@@ -1,17 +1,14 @@
...
     "sourceMap": true,
     "declaration": true,
     "declarationMap": true,
     "resolveJsonModule": true,
-    "strict": true,
-    "moduleResolution": "node",
```
Comment (SimenB) [inline tsconfig.json:13]: comes from extended config

File: tsconfig.json [hunk 3]
```diff
@@ -1,17 +1,14 @@
...
-    "downlevelIteration": true
```
Comment (SimenB) [inline tsconfig.json:14]: why?

File: tsconfig.json [hunk 4]
```diff
@@ -1,17 +1,14 @@
...
+    "esModuleInterop": false,
+    "skipLibCheck": false
```
Comment (SimenB) [inline tsconfig.json:11]: overrides extended config to be stricter (and matches what you had before)

**Review by ivarconr — APPROVED** [review]
Looks good.  Thanks 🙏👍

**Pull request merged**